### ISSUE TITLE
Tool to easily migrate between Aegir Master and Octopus

### ISSUE THREAD

**Issue by chrisarcherhydrant** [omega8cc/boa#1221]
Hi,

While you have a tool to support migration of Octopus instances between remote servers, the tool to migrate from master to Octopus doesn’t exist yet.  

I believe this can be done manually but could not find any documentation around this.

Thanks,
Chris

**Comment by omega8cc** [comment]
Moving platforms and sites from master to satellite on the same system is technically very simple.

But such tool should also support mass imports of platforms and sites from remote classic Aegir systems, and be backward compatible with both Aegir 3.x and 2.x -- and *this* will be pretty tricky, especially on systems with non-standard features enabled.

**Comment by c-archer** [comment]
Is there any documentation on how to do this manually?

**Comment by omega8cc** [comment]
@c-archer No, it's not documented, because it's not supported, and very clearly discouraged practice to use the master instance for anything. That said, we are aware that there are situations where people hire developers who don't bother to read the docs and don't follow the best practices, leaving system owner with unexpected trouble. This is the reason why we plan to support migrations from master to satellite instances via dedicated script, and document only the scope related to this script usage.

In short, the steps to migrate sites between `/var/aegir` and `/data/disk/o1` should look like this:

1. Install Octopus with `boa in-octopus` command (see the `docs/INSTALL.txt` for details)
2. Move entire platform with sites from master to `/data/disk/o1/static/migrated/`
3. Fix initial ownership with `chown -R o1:users /data/disk/o1/static/migrated/platform_name`
4. Add moved platform in the Octopus Aegir interface
5. Once sites are imported and verified, run Registry Rebuild task on all sites
6. Remove duplicate vhosts in the master instance and reload nginx
7. Delete migrated platform node in the master instance (don't run delete tasks!)
8. if any site experiences issues, run cache clear task in Octopus or restart redis-server

As you can see, doing this manually will cause some downtime for the affected sites, while the script should help to minimize the downtime and automate all steps.
